### PR TITLE
Add google/osqp-cpp@0.0.0-20231004-4343373 to BCR

### PR DESCRIPTION
**Pull request by udaya2899**
This currently doesn't build because google/osqp-cpp needs osqp to be built with 64-bit integers. As I understand, https://registry.bazel.build/modules/osqp is only built with 32-bit integers.

### CODE REVIEW

File: modules/osqp-cpp/metadata.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+            "email": "udayaprakash2899@gmail.com",
+            "github": "udaya2899",
+            "name": "Udaya Prakash Nageswaran"
+        }
+    ],
+    "repository": [
+        "github:google/osqp-cpp"
+    ],
+    "versions": [
+        "20231004.0.bcr.1"
```
Comment (mering) [inline modules/osqp-cpp/metadata.json:14]: As this repository doesn't have any Git tags, I suggest using the format `0.0.0-YYYYMMDD-SHORTSHA` (`0.0.0-20240530-2db0eb3`) as is done for other modules.

File: modules/osqp-cpp/metadata.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+    "homepage": "https://github.com/google/osqp-cpp",
+    "maintainers": [
+        {
+            "email": "udayaprakash2899@gmail.com",
+            "github": "udaya2899",
+            "name": "Udaya Prakash Nageswaran"
```
Comment (mering) [inline modules/osqp-cpp/metadata.json:7]: You might want to contact the owners of the upstream repo if they are willing to help with maintaining.

File: modules/osqp-cpp/20231004.0.bcr.1/presubmit.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  - 7.x
+  - 6.x
+tasks:
+  verify_targets:
+    name: Verify build targets
+    platform: ${{ platform }}
+    bazel: ${{ bazel }}
+    build_targets:
+    - '@osqp-cpp//:osqp++'
+    - '@osqp-cpp//:license'
```
Comment (mering) [inline modules/osqp-cpp/20231004.0.bcr.1/presubmit.yml:18]: Consider the wildcard to keep it easier to maintain in the future.

```suggestion
    - '@osqp-cpp//...'
```

File: modules/osqp-cpp/20231004.0.bcr.1/MODULE.bazel [hunk 4]
```diff
@@ -0,0 +1,8 @@
+module(
+    name = "osqp-cpp",
+    version = "20231004.0.bcr.1",
+    compatibility_level = 0,
```
Comment (mering) [inline modules/osqp-cpp/20231004.0.bcr.1/MODULE.bazel:4]: You can leave it unset by default
```suggestion
```

File: modules/osqp-cpp/20231004.0.bcr.1/MODULE.bazel
Comment (mering) [inline modules/osqp-cpp/20231004.0.bcr.1/MODULE.bazel:1]: You need to add a patch for adding this file to the repo as well.

**Comment by bazel-io** [comment]
Hello @wep21, modules you maintain (osqp) have been updated in this PR. Please review the changes.

File: modules/osqp-cpp/0.0.0-20231004-4343373/source.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+    "url": "https://github.com/google/osqp-cpp/archive/43433736334d6b515ea4b0247156fea9e56c0d3f.tar.gz",
+    "integrity": "sha256-Ye1aTRuJMM0T6u6Z/qZLAt8iCEzmRArQVxOUPAHkwSQ=",
+    "strip_prefix": "osqp-cpp-43433736334d6b515ea4b0247156fea9e56c0d3f",
+    "patches": {
+        "add-build-file.patch": "sha256-h4DBMwvSkV+caG2l/KGwBnrrM0UcbdgjEr2Dn9YTZTw=",
+        "add_module_dot_bazel.patch": "sha256-dv+i2dhf/c1zvvR8NWKFMKUSzWVjJdEt5ZoZ8QLtf2s="
+    },
+    "patch_strip": 1
```
Comment (mering) [inline modules/osqp-cpp/0.0.0-20231004-4343373/source.json:9]: Remove this as your patches do not have a prefix
```suggestion
    }
```
```suggestion
    },
    "patch_strip": 1
```

**Comment by mering** [comment]
@fmeum could you add the `skip-url-stability-check` label? Can you help us interpreting the other failure? Thanks.

**Comment by udaya2899** [comment]
Marking the PR as draft till #2147 is merged. I also need to clarify some license notices. I'm able to build this successfully locally though. I'll be back to work on Monday (June 2) to check this. Thanks for the help so far @fmeum

**Comment by udaya2899** [comment]
@fmeum It's now ready for review, please check. Thanks :)

**Review by fmeum — APPROVED** [review]

**Pull request merged**